src/kOS.Safe/Utilities/StringUtil.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+namespace kOS.Safe
+{
+    /// <summary>
+    /// String utility functions that avoid the culture-aware overhead of the builtin string functions.
+    /// </summary>
+    public static class StringUtil
+    {
+        public static bool EndsWith(string str, string suffix)
+        {
+            int strLen = str.Length;
+            int suffixLen = suffix.Length;
+
+            if (strLen < suffixLen)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < suffixLen; i++)
+            {
+                if (str[strLen - i - 1] != suffix[suffixLen - i - 1])
```

Comment:
I'm going to switch to counting this forward to save some math.  (set index to strLen - i - 1 at first, then just increment it by +1 each iteration).